### PR TITLE
Change logic of deleting RBS service

### PR DESCRIPTION
**Pull request by panslava**
- Delete only if service has finalizer or rbs forwarding rule, meaning it was provisioned by RBS controller at some point
- Don't delete RBS annotation on garbage collection

/assign @cezarygerard @aojea

### CODE REVIEW

File: pkg/l4lb/l4netlbcontroller.go [hunk 1]
```diff
@@ -621,14 +622,14 @@ func (lc *L4NetLBController) garbageCollectRBSNetLB(key string, svc *v1.Service)
 
 	// IMPORTANT: Remove LB annotations from the Service LAST, cause changing service fields are emitted as service update to other controllers
 	if lc.enableDualStack {
-		if err := deleteL4RBSDualStackAnnotations(lc.ctx, svc); err != nil {
```
Comment (aojea) [inline pkg/l4lb/l4netlbcontroller.go:624]: This is the same error we found yesterday, the last thing to be removed should be the finalizer, also why do we need to remove the annotations if it is going to be deleted?

Comment (aojea) [reply pkg/l4lb/l4netlbcontroller.go:624]: should not we just delete the finalizer and bail out?

Comment (panslava) [reply pkg/l4lb/l4netlbcontroller.go:624]: Annotation deletion covers the case of garbage collection on switching service types (to clusterIP or ILB), when service not really completely deleted

Comment (panslava) [reply pkg/l4lb/l4netlbcontroller.go:624]: But we should not remove RBS annotation (as it was set up by user) so this PR removes deleting RBS annotation

Comment (aojea) [reply pkg/l4lb/l4netlbcontroller.go:624]: >  switching service types (

ok, but then we should have 2 flows , one for deletion and other for mutation.
The deletion just removes the finalizer
The mutation just removes the annotations added by the controller but keeps the finalizer

IIUIC this still conflates both

**Pull request merged**